docs/metatype.dev/docs/reference/policies/index.mdx

```diff
@@ -28,3 +28,33 @@ Policies are hierarchical in the sense that the request starts with a denial, an
   typescript={require("!!code-loader!../../../../../examples/typegraphs/policies.ts")}
   query={require("./policies.graphql")}
 />
+
+## Composition rules
+
+### Traversal order
+
+- `ALLOW`: Allows access to the parent and all its descendants, disregarding inner policies.
+- `DENY`: Denies access to the parent and all its descendants, disregarding inner policies.
+- `PASS`: Allows access to the parent, each descendant will still be evaluated individually (equivalent to having no policies set).
+
+### Inline chain
+
+If you have `foo.with_policy(A, B).with_policy(C)` for example, it will be merged into a single chain `[A, B, C]`.
+
+The evaluation is as follows:
+
+- `ALLOW` and `DENY` compose the same as `true` and `false` under the logical `AND` operator.
+- `PASS` does not participate.
+
+Or more concretely:
+
+- `ALLOW` & Other -> Other
```

Comment:
This might be misleading, no? Doens't `ALLOW` completely short-circuit policy evaluation? I.e., if we get an allow, do we evaluate other members of the policy set on that node? I do like the analogy but maybe you should note this fact.

Reply:
Also, we should not that ALLOW would be a very rare case. Most policies would not want to override any possible sibling policies. I'd avoid the alphanumeric ordering and list allow last to highlight this fact.

Reply:
>  Doens't ALLOW completely short-circuit policy evaluation

In traversal order yes. 

But since we allow multiple policies to be put on a single type I figured just `AND`-ing them would be more natural.

In fact, it follows the idea of...
```ts
struct(
  a: integer().with_policy(A),
  b: integer().with_policy(B)
)
```

The policies would be hoisted as `[A, B]`, and field access is just `A & B` if we assume we project all fields. Thus by extension...
```ts
struct(
  a: integer().with_policy(A),
  b: integer().with_policy(B, X)
)
```
We have `[A, B, X] := A & B & X`.

Reply:
I have reformulated the explanation into something like "if one item on the chain denies then the entire thing denies".

Still the same idea but hopefully more simple to reason about.

E.g. `pseudo: t.string().with_policy(deny).with_policy(allow)` should deny `pseudo`

Reply:
@Yohe-Am 

I have reformulated the explanation into something like "if one item on the chain denies then the entire thing denies".

Still the same idea but hopefully more simple to reason about.

E.g. `pseudo: t.string().with_policy(deny).with_policy(allow)` should deny..